swift/apple/FirezoneKit/Sources/FirezoneKit/Features/SettingsView.swift

```diff
@@ -154,6 +188,97 @@ public struct SettingsView: View {
     model.load()
     dismiss()
   }
+
+#if os(macOS)
+  func exportLogsButtonTapped() {
+    self.isExportingLogs = true
+
+    let savePanel = NSSavePanel()
+    savePanel.prompt = "Save"
+    savePanel.nameFieldLabel = "Save log zip bundle to:"
+    savePanel.nameFieldStringValue = "firezone-logs.zip"
+
+    guard let window = NSApp.windows.first(where: { $0.identifier?.rawValue.hasPrefix("firezone-settings") ?? false }) else {
+      self.isExportingLogs = false
+      logger.log("Settings window not found. Can't show save panel.")
+      return
+    }
+
+    savePanel.beginSheetModal(for: window) { response in
+      guard response == .OK else {
+        self.isExportingLogs = false
+        return
+      }
+      guard let destinationURL = savePanel.url else {
+        self.isExportingLogs = false
+        return
+      }
+
+      Task {
+        do {
+          try await createLogZipBundle(destinationURL: destinationURL)
+          self.isExportingLogs = false
+          await MainActor.run {
+            window.contentViewController?.presentingViewController?.dismiss(self)
+          }
+        } catch {
+          self.isExportingLogs = false
+          await MainActor.run {
+            // Show alert
+          }
+        }
+      }
+    }
+  }
+#elseif os(iOS)
+  func exportLogsButtonTapped() {
+    self.isExportingLogs = true
+    Task {
+      let logsTempZipFileURL = try await createLogZipBundle()
+      self.isPresentingExportLogShareSheet = true
+    }
+  }
+#endif
+
+  @discardableResult
+  private func createLogZipBundle(destinationURL: URL? = nil) async throws -> URL {
+    let fileManager = FileManager.default
+    guard let logFilesFolderURL = SharedAccess.logFolderURL else {
+      throw SettingsViewError.logFolderIsUnavailable
+    }
+    let zipFileURL = destinationURL ?? fileManager.temporaryDirectory.appendingPathComponent("firezone_logs.zip")
+    if fileManager.fileExists(atPath: zipFileURL.path) {
+      try fileManager.removeItem(at: zipFileURL)
+    }
+    let task = Task.detached(priority: .userInitiated) { () -> URL in
+      try FileManager.default.zipItem(at: logFilesFolderURL, to: zipFileURL)
```

Comment:
Should we remove the old log files once the logFile has been successfully written? Or since these are in a cache directly will they be automatically garbage-collected?

Reply:
The app doesn't do anything to the log files after zipping them up. They might get uploaded to the server later on, or not. I assumed that cleaning up connlib's logs is connlib's responsibility. Should the app remove them? (We don't write app logs to a file yet, though the directory structure has been made with that in mind.)

We do write the logs to a cache directory, so the OS can reclaim the space when under disk-space-pressure.

Reply:
@roop Perfect, that works!